sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -24,7 +33,9 @@ def test_transform_linear_model():
                 X = func(iris.data)
                 clf.set_params(penalty="l1")
                 clf.fit(X, iris.target)
-                X_new = clf.transform(X, thresh)
+                clean_warning_registry()
+                with warnings.catch_warnings(record=True) as record:
```

Comment:
assert_warns.

I note that all these tests will disappear with deprecation... a little comment to that effect might be helpful for maintenance.